### DISCUSSION TITLE
Issue with Item Fufillment creation, and commit quantities (only with inventoryLocation)

### DISCUSSION THREAD

**Discussion by benjy8001** [netsuitephp/netsuite-php#169] [Q&A] [answered]
Hi !

Actually i create an Item Fufillment from a Sales Order (with an intialize request and a create request), but each line items have no inventory location set.

When i try to create an item fulfillment from a sales order including items with inventory location, it leads to this message :

`App\Exceptions\Netsuite\ItemFulfillmentCreationException  Unable to create Item Fulfillment from Sale Order 5278060
You must have at least one valid line item for this transaction.`

My goal is to : 
1/ Create an item fulfillment linked to a Sales order
2/ Commit quantities in the location in order to have an up to date inventory.

Maybe the `ItemFulfillmentItem` class is missing the `inventoryLocation` properties ?

Thanks for your help !

**Comment by benjy8001** [comment]
I see this answer https://stackoverflow.com/a/59401242/3022243
But how to use "transform" request and pass a default value ?

**Reply by jrebs** [reply]
> I see this answer https://stackoverflow.com/a/59401242/3022243 But how to use "transform" request and pass a default value ?

Assuming transform is another term for initialize, look in the examples provided by this package.

https://github.com/netsuitephp/netsuite-php/blob/master/EXAMPLES.md#Add-an-item-fulfillment

**Reply by benjy8001** [reply]
Thanks, this is exactly what i'm doing.

Below the body send to Netsuite : 

```
<SOAP-ENV:Body>
<ns2:initialize>
<ns2:initializeRecord>
<ns1:type>itemFulfillment</ns1:type>
<ns1:reference internalId="5278060" type="salesOrder"/>
</ns2:initializeRecord>
</ns2:initialize>
</SOAP-ENV:Body>
```

**Reply by jrebs** [reply]
Initialize request looks fine, so there must be something wrong configuration-wise. NetSuite's error responses are not always the most clear about what the actual problem. A lot of different issues can cause a request to fail without getting back a really logical explanation as to why. Examples are products that are marked as inactive or products that have no inventory in netsuite. In any case, there appears to be nothing wrong with the toolkit, as all it does is generate the soap requests and parse the responses and that all looks normal.

**Reply by benjy8001** [reply]
I agree. I see this also ; https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_1515703010.html#To-enable-Intercompany-Cross-Subsidiary-Fulfillment%3A

I think it's link to this, but if someone have already solve this issue :)

**Comment by jrebs** [comment]
The error message you're getting appears to be an application level message and not something coming directly from this toolkit. `App\Exceptions\Netsuite\ItemFulfillmentCreationException` isn't even part of this package. It would help if you provided more context like exactly what your request looks like and exactly what you're getting back from the web services (you can enable logging or look in the WS log in NetSuite's UI).

**Reply by benjy8001** [reply]
Yes, of course, here the response from Netsuite : 
```
<soapenv:Envelope xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
<soapenv:Header>
<platformMsgs:documentInfo xmlns:platformMsgs="urn:messages_2023_1.platform.webservices.netsuite.com">
<platformMsgs:nsId>WEBSERVICES_4198569_SB1_052820241058709462889998783_70b087a2</platformMsgs:nsId>
</platformMsgs:documentInfo>
</soapenv:Header>
<soapenv:Body>
<initializeResponse xmlns="urn:messages_2023_1.platform.webservices.netsuite.com">
<readResponse>
<platformCore:status xmlns:platformCore="urn:core_2023_1.platform.webservices.netsuite.com" isSuccess="false">
<platformCore:statusDetail type="ERROR">
<platformCore:code>VALID_LINE_ITEM_REQD</platformCore:code>
<platformCore:message>You must have at least one valid line item for this transaction.</platformCore:message>
</platformCore:statusDetail>
</platformCore:status>
</readResponse>
</initializeResponse>
</soapenv:Body>
</soapenv:Envelope>
```

**Comment by benjy8001** [comment, marked as answer]
For notice, passing inventoryLocation field is not possible in Item fulfillment via Soap request and is not supported. In my case, i must use restlet for passing this value like this :

```
var itemFulfillment = record.transform({
            fromType: record.Type.SALES_ORDER,
            fromId: soId,
            toType: record.Type.ITEM_FULFILLMENT,
            isDynamic: true,
            defaultValues: {
                inventorylocation: invLocation
            }
        });
```

**Reply by jrebs** [reply]
The SalesOrder supports a `location` property which references a location defined in Setup > Company > Classifications > Locations. Is this what you would have been intending to set for the SOAP call? Are they the same property having different names in the REST calls or is this actually a different value with different meaning?
